### PR TITLE
Add ID to Cluster API

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:
This PR propose the API change of [clusterid](https://github.com/karmada-io/karmada/tree/master/docs/proposals/clusterid) proposal.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
The `Cluster` API introduced an optional field named `ID` to uniquely identify the cluster.
```

### CODE REVIEW

File: pkg/apis/cluster/types.go [hunk 1]
```diff
@@ -26,6 +26,19 @@ type Cluster struct {
...
+	// - generated by command 'karmadactl join' for clusters in Push mode.
+	// - generated by `karmada-agent` for clusters in Pull mode.
+	//
+	// It is not recommended to update this value unless you really know what you are doing, because
+	// it will/may be used to :
+	// - uniquely identify the clusters within the Karmada system.
+	// - compose the DNS name of multi-cluster services.
+	//
+	// +optional
+	ID string `json:"id,omitempty"`
```
Comment (GitHubxsy) [inline pkg/apis/cluster/types.go:40]: What is the difference between metadata.uid and cluster id?

**Review by GitHubxsy — CHANGES_REQUESTED** [review]

File: pkg/apis/cluster/types.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/apis/cluster/types.go:39]: Updated.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Pull request merged**